src/main/java/soolutions/hometaskapp/basic/BasicUser.java

```diff
@@ -0,0 +1,6 @@
+package soolutions.hometaskapp.basic;
+
+import soolutions.hometaskapp.User;
+
+public class BasicUser implements User {
```

Comment:
Is User and BasicUser a necessary abstraction? Would the system be worse if we just had a concrete class for User?

Reply:
@SilasReinagel in financial context, don't they typically deal with legacy system integrations and such? When a bank merges with another bank?

In this context, I think having `User` as an interface is a good choice. Otherwise, `User` would have to be subclassed which is the strongest form of coupling there is.

Reply:
What if we want to have authentication and authorisation in the system?

Reply:
That can happen as a cross-cutting concern, or via decoration. Authorization only matters at the periphery of the system.